ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-igmp.j2

```diff
@@ -7,7 +7,19 @@
 {% if router_igmp is arista.avd.defined %}
 !
 router igmp
+{%     if router_igmp.host_proxy_match_mroute is arista.avd.defined %}
+   host-proxy match mroute {{ router_igmp.host_proxy_match_mroute }}
+{%     endif %}
 {%     if router_igmp.ssm_aware is arista.avd.defined(true) %}
    ssm aware
 {%     endif %}
+{%     if router_igmp.vrfs is arista.avd.defined %}
+{%         for vrf in router_igmp.vrfs | arista.avd.natural_sort('name') %}
+{%             if vrf.name != 'default' and vrf.host_proxy_match_mroute is arista.avd.defined %}
+   !
+   vrf {{ vrf.name }}
+     host-proxy match mroute {{ vrf.host_proxy_match_mroute }}
+{%             endif %}
+{%         endfor %}
+{%     endif %}
```

Comment:
```suggestion
{%     for vrf in router_igmp.vrfs | arista.avd.natural_sort('name') %}
{%         if vrf.host_proxy_match_mroute is arista.avd.defined %}
   !
   vrf {{ vrf.name }}
     host-proxy match mroute {{ vrf.host_proxy_match_mroute }}
{%         endif %}
{%     endfor %}
```

Reply:
I added the "clever" logic because router igmp does not move the config back to the global context but spits out an error if you try to define vrf default :

![image](https://github.com/aristanetworks/avd/assets/8918617/c875f983-265c-47b7-99f2-f9042362bdb1)

Reply:
I'm afraid this could lead to people thinking there is a bug since if you input vrf default, it simply doesn't output anything. Nothing warns the user that it is incorrect.

Reply:
I would rather the user see the message of invalid config coming from EOS. In eos_designs we could offer such logic to protect the user!

Reply:
Also `{%         if vrf.host_proxy_match_mroute is arista.avd.defined %}` should be moved under `vrf {{ vrf.name }}` since there are multiple options under this cli:

```
eos-1(config-router-igmp-vrf-test)#sh active 
router igmp
   vrf test
      host-proxy match mroute iif
      ip igmp access-group test1
```

Reply:
We don't support other options now, so we should not implement that in the template.

Reply:
moved the vrf creation above the match mroute if loop